src/arith/canonical_simplify.cc

```diff
@@ -77,6 +77,25 @@ inline PrimExpr DivImpl(PrimExpr a, PrimExpr b, DivMode mode) {
   }
 }
 
+bool CheckCastImpl(DataType dtype, PrimExpr value, Analyzer* analyzer) {
```

Comment:
Perhaps rename to UpcastIsSafe?

Reply:
document this function

Reply:
I guess CastIsSafe seems better, since it checks both upcast and downcast